wagtail/contrib/search_promotions/templates/wagtailsearchpromotions/list.html

```diff
@@ -32,7 +32,11 @@
                 </td>
                 <td>
                     {% for searchpick in query.editors_picks.all %}
-                        <a href="{% url 'wagtailadmin_pages:edit' searchpick.page.id %}" class="nolink">{{ searchpick.page.title }}</a>{% if not forloop.last %}, {% endif %}
+                        {% if searchpick.recommendation.page %}
+                            <a href="{% url 'wagtailadmin_pages:edit' searchpick.page.recommendation.id %}" class="nolink">{{ searchpick.title }}</a>{% if not forloop.last %}, {% endif %}
+                        {% else %}
+                            <a href="{{ searchpick.recommendation.external_link }}" class="nolink">{{ searchpick.title }}</a>{% if not forloop.last %}, {% endif %}
+                        {% endif %}
```

Comment:
Not sure why but these links are not working for me at all, the href is blank, can you please review.

<img width="1278" alt="Screenshot 2023-08-24 at 8 14 20 am" src="https://github.com/wagtail/wagtail/assets/1396140/2725c49d-9aff-4e3d-a18b-054bcf283992">

I made one with a link to google and the other with a Page and the href is just empty for both.